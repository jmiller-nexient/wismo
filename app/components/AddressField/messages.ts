import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.address-field';

export default defineMessages({
  addressFieldLabel: {
    id: `${scope}.addressFieldLabel`,
    defaultMessage: 'Address',
  },
});
