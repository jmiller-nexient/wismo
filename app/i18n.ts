/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 *   IMPORTANT: This file is used by the internal build
 *   script `extract-intl`, and must use CommonJS module syntax
 *   You CANNOT use import/export in this file.
 */
import { DEFAULT_LOCALE } from './locales';

const enTranslationMessages = require('translations/en.json');
const esTranslationMessages = require('translations/es.json');

// The purpose of this method is to apply fallback logic when a translation
// is not defined for the current locale.
// In our case, we do not want any fallback logic.
// tslint:disable-next-line: typedef
const formatTranslationMessages = (locale: string, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages) : {};

  // tslint:disable-next-line: typedef
  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE ? defaultFormattedMessages[key] : messages[key];

    return { ...formattedMessages, [key]: formattedMessage };
  };

  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
  en: formatTranslationMessages('en', enTranslationMessages),
  es: formatTranslationMessages('es', esTranslationMessages),
};
