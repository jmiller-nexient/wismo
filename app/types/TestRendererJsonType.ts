import { ReactTestRendererJSON } from 'react-test-renderer';

type TestRendererJsonType = ReactTestRendererJSON | null;

export default TestRendererJsonType;
