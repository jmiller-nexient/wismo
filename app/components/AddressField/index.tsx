import React from 'react';

import Address from 'components/Address';
import BaseFormField from 'components/BaseField';

import IAddress from 'types/address';

import messages from './messages';

interface IAddressFieldProps {
  address: IAddress;
}

const AddressField: React.FC<IAddressFieldProps> = (props: IAddressFieldProps): React.ReactElement => {
  const { address } = props;

  return (
    <BaseFormField label={messages.addressFieldLabel}>
      <Address {...address}/>
    </BaseFormField>
  );
};

export default AddressField;
