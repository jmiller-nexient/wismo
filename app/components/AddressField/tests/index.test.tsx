import React from 'react';
import renderer from 'react-test-renderer';

import { IntlProvider } from 'react-intl';

import { translationMessages } from '../../../i18n';

import AddressField from 'components/AddressField';

import TestRendererJsonType from 'types/TestRendererJsonType';

const address = {
  line1: '1234 That Street',
  line2: null,
  city: 'Anytown',
  state: 'MI',
  zip: '12345'
};

describe('<AddressField />', () => {
  it('should render correctly when a valid address is passed in', () => {
    const tree: TestRendererJsonType = renderer.create(
      <IntlProvider locale="en" messages={translationMessages.en}>
        <AddressField address={address}/>
      </IntlProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
