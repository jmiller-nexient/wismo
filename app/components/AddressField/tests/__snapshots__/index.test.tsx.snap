// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressField /> should render correctly when a valid address is passed in 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Address
  </label>
  <div>
    <span
      className="dte-wismo-base-field-children"
    >
      <address
        className="dte-wismo-address"
      >
        1234 That Street
        <br />
        Anytown, MI 12345
      </address>
    </span>
  </div>
</div>
`;
